imports/api/PointCalculationUtils.js

```diff
@@ -0,0 +1,132 @@
+import { Constants } from './Constants';
+import { NewGameUtils } from './NewGameUtils';
+
+export var PointCalculationUtils = {
+	jpn_dealin_delta,
+	jpn_selfdraw_delta,
+	jpn_mistake_delta
+};
+
+/**
+ * Calculates the change in points as a result of a deal-in hand
+ * Uses the formula from https://en.wikipedia.org/wiki/Japanese_Mahjong_scoring_rules
+ * 
+ * @return {Object} containing the point difference for each seat
+ */
+function jpn_dealin_delta(points, fu, winnerWind, loserWind, riichiSticks) {
+	let winds = {};
+	winds[Constants.EAST] = winds[Constants.SOUTH] = winds[Constants.WEST] = winds[Constants.NORTH] = 0;
+
+	let basicPoints;
+	let currentBonus = Number(Session.get("current_bonus")) * Constants.BONUS_POINTS;
+
+	// The multiplier is for whether or not it's a dealer victory
+	let multiplier = (winnerWind != NewGameUtils.roundToDealerDirection(Number(Session.get("current_round")))) ? 4 : 6;
+
+	// Check to see if you have to count basic points
+	if (points < 5) {
+		if (fu == 20 || (points == 1 && fu == 25)) {
+			return 0; // Issue Protection
+		} else {
+			let manganPayout = Constants.JPN_MANGAN_BASE_POINTS * multiplier;
+			basicPoints = Math.ceil((fu * Math.pow(2, 2 + points)) * multiplier / 100) * 100;
+			basicPoints = basicPoints < manganPayout ? basicPoints : manganPayout;
+		}
+	} else { basicPoints = manganValue(points) * multiplier };
```

Comment:
If you're going to do this, the semicolon should be at the end of the line, not the end of the block.  I would still make this and the other one like it a seperate line.